viewer-elm/src/Fps.elm

```diff
@@ -1,11 +1,26 @@
-module Fps exposing (showAverage, update)
+module Fps exposing (init, showAverage, update)
 
--- FPS logic copied from https://github.com/w0rm/elm-physics/blob/master/examples/Common/Fps.elm
+-- FPS logic inspired by https://github.com/w0rm/elm-physics/blob/master/examples/Common/Fps.elm
 
 
-update : Float -> List Float -> List Float
-update dt fps =
-    List.take 50 (dt :: fps)
+update : Bool -> Float -> List Float -> List Float
+update hasNewFrame dt fps =
+    List.take 50 <|
+        if hasNewFrame then
```

Comment:
The key change in FPS logic: if the exact frame is available, add (as in list cons) time delta separately, otherwise add it (as in `+`) to the previous delta.